themes/default/content/blog/iac-recommended-practices-structuring-pulumi-projects/index.md

```diff
@@ -0,0 +1,104 @@
+
+## Moving forward with Zephyr
+
+Zephyr had a number of factors in play, not the least of which was the organizational change resulting from their growth and success. In the end, Zephyr decided to restructure their use of Pulumi for the online retail store into three separate projects:
+
+* An infrastructure project: This project handled the AWS VPC and related infrastructure resources.
+* A platform project: This project managed the Kubernetes cluster(s).
+* An application project: This project handled the actual online retail store itself.
+
+{{% notes %}}
+You can use a combination of code refactoring and `pulumi import` to assist in the process of restructuring Pulumi projects, but it some cases it may be easier to create new infrastructure and migrate over to it.
+{{% /notes %}}
+
+In [the first Zephyr post](/blog/iac-recommended-practices-code-organization-and-stacks/), you read about some of the options for where to store Pulumi code relative to the applications it supports. At that time, Zephyr had decided to go with what's called a monorepo approach, meaning that the Pulumi code was stored in the same repository as the application code. In conjunction with the project reorganization, Zephyr also decided to change how their code was organized:
+
+* Code for the infrastructure project moved to a separate "zephyr-infra" repository
+* A new "zephyr-k8s" repository housed the code for the platform project
+* The code for the online retail store application itself remained in the same "zephyr-app" repository
+
+Note that relocating code (to a different GitHub repository or different filesystem location within a repository) generally has no impact on Pulumi. As long as the project name and stack name do not change, then changing the filesystem location won't affect anything---keeping in mind, of course, that filesystem paths and such may need to be adjusted in code to account for any such changes.
+
+These repositories are available for you to review on GitHub; use the branch/tag selector to find the `multi-project` tab in each repository to see the state of the code as of this blog post.
+
+## Summarizing recommended practices
+
+This post covered the following recommended practices for working with Pulumi:
+
+* **Use multiple Pulumi projects** when you need the added flexibility that multiple Pulumi projects offer.
+* **Align your Pulumi projects** according to external factors like company/department/team structure, resource dependencies, lifecycle, and use case/application.
+* **Change the filesystem location** for Pulumi code as needed without impacting the resources managed by that code, as long as the Pulumi project name and stack name don't change.
+
+The next post will focus on some specific technical details related to Zephyr's project reorganization; specifically, how they used StackReferences to share information between multiple Pulumi projects. Stay tuned!
```

Comment:
Since we're talking specifically about these projects:

```suggestion
The next post will focus on some specific technical details related to Zephyr's project reorganization; specifically, how they used StackReferences to share information between these Pulumi projects. Stay tuned!
```